src/stats.ts

```diff
@@ -0,0 +1,121 @@
+import { Client } from './client';
+import { GenericObject, StatsEvent } from './types';
+import { isDfid } from './util/is';
+
+export class StatsClient extends Client {
```

Comment:
For the code you propose maybe we should use the `clientRepo` for that, as it is the one keeping a pool of clients to use, and I'd have to have a client object inside to call here and there and making the requests. 

Maybe, just maybe, we can refactor the common code (hashid, request method, and so on) and make each a child of this new parent class. Let me know what you think.

Reply:
`ClientRepo` is a tool you can use if you want, but we should not couple code to it. The same with client repo, for me, would be:

```js
import { repo } from './myrepo';

const stats = new StatsClient(repo.getClient());
// …
repo.zone = Zone.US;
stats.use(repo.getClient());
```

But now that you mention it, the API would be nicer if we do:

```js
import { repo } from './myrepo';

const stats = new StatsClient(repo.getClient(Zone.EU));
// …
stats.use(repo.getClient(Zone.US));
```

Reply:
Anyway, we are here duplicating the client, and in my code we are just reusing. Have you considered the refactor of common code?

Reply:
What are we duplicating here…? You have a wrapper for the client to perform stats calls easily. This object just handles a reference to an instance of `Client`.

For me this would be even more decoupled if the client doesn't hold the `hashid` internally (in that case, to decouple even more, my proposal of reducing parameters by storing the `sessionId` is not valid - see in a different comment).

- `Client` provides methods to call certain endpoints (search, options, stats) with basic parameters, but it doesn't know all the parameters for all the endpoints, just the basic ones.
- `Query` provides a way to configure parameters for the search endpoint in an easy way.
- `StatsClient` provides a wrapper to perform frequently done stats calls in an easy way.

They are different tools but I don't see the duplication here, I see composition over inheritance.

We can discuss whether to add these methods to the Client class, so it's available directly, but I think inheritance is not the way to go.

Composition lets me add more stuff specific to stats without messing with the client, so it remains small and focused on making requests to valid endpoints.